sdk/core/azure-core/src/main/java/com/azure/core/util/AzureUserAgentUtil.java

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.util.logging.ClientLogger;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Utility class for fetching the user agent properties like client library name and version.
+ */
+public class AzureUserAgentUtil {
+
+    private static final String UNKNOWN_VERSION = "UnknownVersion";
+    private static final String UNKNOWN_NAME = "UnknownName";
+    private static final String VERSION_PROPERTY = "version";
+    private static final String NAME_PROPERTY = "name";
+    private static final Map<String, UserAgentProperties> USER_AGENT_PROPERTIES = new ConcurrentHashMap<>();
+
+    private AzureUserAgentUtil() {
+        // no instances for this class
+    }
+
+    /**
+     * Helper method that returns the {@link UserAgentProperties} defined in the {@code propertiesFileName}.
+     *
+     * @param propertiesFileName The file name defining the user agent properties.
+     * @return the {@link UserAgentProperties}.
+     */
+    public static UserAgentProperties getUserAgentProperties(String propertiesFileName) {
+        return USER_AGENT_PROPERTIES.computeIfAbsent(propertiesFileName, AzureUserAgentUtil::readFromPropertiesFile);
+    }
+
+    private static UserAgentProperties readFromPropertiesFile(String propertiesFileName) {
```

Comment:
I would be inclined to just return a Properties or Map, rather than have a new type.

Reply:
I thought about returning a Map/Properties but that would require each of the client libraries to know/define the keys ("name", "version" etc). 

`Properties` type is mutable, so, Map may be a better option here if a new type is not preferable.

Reply:
Returning a `Map<String, String>` now.